internal/controllers/machineset/machineset_controller.go

```diff
@@ -397,6 +398,42 @@ func (r *Reconciler) syncMachines(ctx context.Context, machineSet *clusterv1.Mac
 			return errors.Wrapf(err, "failed to update Machine %q", klog.KObj(updatedMachine))
 		}
 		machines[i] = updatedMachine
+
+		infraMachine, err := external.Get(ctx, r.Client, &updatedMachine.Spec.InfrastructureRef, updatedMachine.Namespace)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get Infrastructure Machine %s",
+				klog.KRef(updatedMachine.Spec.InfrastructureRef.Namespace, updatedMachine.Spec.InfrastructureRef.Name))
+		}
+		// Cleanup managed fields of all Infrastructure Machines.
+		// We do this so that Infrastructure Machines that are created using the Create method can also work with SSA.
+		// Otherwise, labels and annotations would be co-owned by our "old" "manager" and "capi-machineset" and then we
+		// would not be able to e.g. drop labels and annotations.
+		if err := ssa.CleanUpManagedFieldsForSSAAdoption(ctx, infraMachine, machineSetManagerName, r.Client); err != nil {
```

Comment:
Let's discuss. I would prefer cleaning up only labels & annotations

Prior art ~ https://github.com/kubernetes-sigs/cluster-api/blob/4d3b1817e3ee53e34cef21575bb29ec2575b4744/internal/controllers/topology/cluster/structuredmerge/dryrun.go#L195-L202

Reply:
I would move filterintent.go into the ssa util package